assignments/week_3/exercise_3_2/exercise_3_2_answer.cpp

```diff
@@ -0,0 +1,93 @@
+#include "exercise_3_2.hpp"
+#include <iostream>
+
+LinkedList::LinkedList() {
+    head = NULL;
+}
+
+void LinkedList::add_to_front(std::string name) {
+    Node *new_node = new Node();
+    new_node->name = name;
+    new_node->next = head;
+    head = new_node;
+    return;
+}
+
+void LinkedList::add_to_back(std::string name){
+    Node *new_node = new Node();
+    new_node->name = name;
+    new_node->next = NULL;
+    if (head != NULL) {
+        Node *curr = head;
+        while (curr->next != NULL) {
+            curr = curr->next;
+        }
+        curr->next = new_node;
+    } else {
+        head = new_node;
+    }
+    return;
+}
+
+void LinkedList::add_at_index(std::string name, int index) {
+    Node *new_node = new Node();
```

Comment:
We don't want to teach them to use `new` like this. I think probably the sanest option is to just make each `next` pointer a `unique_ptr`, because it's a singly-linked-list.

Reply:
If code academy teaches new then I think this is fine, but we have to have the second half where we explain why not to do this. I don't like just yeeting in unique_ptr because it is the correct thing to do as an experienced C++ programmer, they aren't there yet. I am okay letting smaller things like this slide earlier on to reduce complexity as long as we follow up with best practice later on.